nav2_route/src/plugins/graph_file_savers/geojson_graph_file_saver.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2024 John Chrosniak
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "nav2_route/plugins/graph_file_savers/geojson_graph_file_saver.hpp"
+
+namespace nav2_route
+{
+
+void GeoJsonGraphFileSaver::configure(
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr node)
+{
+  RCLCPP_INFO(node->get_logger(), "Configuring geojson graph file saver");
+  logger_ = node->get_logger();
+}
+
+bool GeoJsonGraphFileSaver::saveGraphToFile(
+  Graph & graph, std::string filepath)
+{
+  if (filepath.empty()) {
+    RCLCPP_ERROR(logger_, "File path is empty");
+    return false;
+  }
+  Json json_graph, json_crs, json_properties;
+  json_graph["type"] = "FeatureCollection";
+  json_graph["name"] = "graph";
+  json_properties["name"] = "urn:ogc:def:crs:EPSG::3857";
+  json_crs["type"] = "name";
+  json_crs["properties"] = json_properties;
+  json_graph["crs"] = json_crs;
+  std::vector<Json> json_features;
+  try {
+    loadNodesFromGraph(graph, json_features);
+    loadEdgesFromGraph(graph, json_features);
+    json_graph["features"] = json_features;
+    std::ofstream file(filepath);
+    file << json_graph.dump(4) << std::endl;
+    file.close();
+  } catch (const std::exception & e) {
+    RCLCPP_ERROR(logger_, "An error occurred: %s", e.what());
+    return false;
+  }
+  return true;
+}
+
+void GeoJsonGraphFileSaver::loadNodesFromGraph(
+  Graph & graph, std::vector<Json> & json_features)
+{
+  for (const auto & node : graph) {
+    Json json_feature, json_properties, json_geometry, json_metadata, json_operations;
+    json_geometry["type"] = "Point";
+    json_geometry["coordinates"] = std::vector<float>{node.coords.x, node.coords.y};
+    json_feature["geometry"] = json_geometry;
+    json_properties["id"] = node.nodeid;
+    json_properties["frame"] = node.coords.frame_id;
+    convertMetaDataToJson(node.metadata, json_metadata);
+    if (json_metadata.size()) {
```

Comment:
```suggestion
    if (json_metadata.size() > 0) {
```

Just to be more explicit